api/types/types_drivers_executor.go

```diff
@@ -132,6 +158,7 @@ type ProvidesStorageExecutorCLI interface {
 // the driver implementations of the StorageExecutor interface.
 type StorageExecutorCLI interface {
 	StorageExecutorFunctions
+	StorageExecutorWithSupported
```

Comment:
@codenrhoden,

Ah, yes, another good question. These [two lines](https://github.com/akutz/libstorage/blob/bugfix/missing-client-support/api/types/types_drivers_executor.go#L160-L161) are your answer. Go can build cumulative interface definitions, but it cannot handle if the same function (or functions) exists in two of them.

So since the `StorageExecutorCLI` interface has a cumulative base of `StorageExecutorFunctions` and `StorageExecutorWithSupported`, if the latter inherited the former `StorageExecutorCLI` would not be able to "inherit" from them both. Try it and see what happens :)

![image](https://cloud.githubusercontent.com/assets/101085/19328611/104c51f8-909a-11e6-97ef-6b98020e275a.png)

```bash
[0]akutz@pax:types$ go build
# github.com/emccode/libstorage/api/types
./types_drivers_executor.go:175: duplicate method InstanceID
./types_drivers_executor.go:175: duplicate method LocalDevices
./types_drivers_executor.go:175: duplicate method NextDevice
```